srt-protocol/src/crypto.rs

```diff
@@ -270,6 +270,68 @@ impl CryptoManager {
 
         ret
     }
+
+    pub fn rekey(&mut self, kmreq: &SrtKeyMessage) -> Result<(), ConnectionReject> {
+        assert_eq!(kmreq.pt, PacketType::KeyingMaterial);
+
+        let salt = kmreq.salt[..].try_into().unwrap();
+        let kek = CryptoManager::gen_kek(&self.options, &salt);
+
+        assert_eq!(
+            kmreq.wrapped_keys.len(),
+            kmreq.key_flags.bits().count_ones() as usize * usize::from(self.options.size) + 8
+        );
+
+        let mut keys = vec![0; kmreq.wrapped_keys.len() - 8];
+
+        let mut iv = [0; 8];
+        match kek.len() {
```

Comment:
Would be great if you could extract this match (that's duplicated in `new_from_kmreq`) into something like `fn unwrap_key(kek: &[u8], wrapped_keys: &[u8]) -> Result<Vec<u8>, ConnectioReject>`